scripts/compile_version_upload_s3.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+#
+#
+# Script to compile a languge version and upload it to the cache.
+#
+# This script automates the process to build and upload a Python/Node/Rust/Go
+# version and upload it to S3 making it available for the builders. When a
+# pre-compiled version is available in the cache, builds are faster because they
+# don't have to donwload and compile.
+#
+# LOCAL DEVELOPMENT ENVIRONMENT
+# https://docs.readthedocs.io/en/latest/development/install.html
+#
+# You can run this script from you local environment to create cached version
+# and upload them to MinIO. For this, it's required that you have the MinIO
+# instance running before executing this script command:
+#
+#   inv docker.up
+#
+#
+# PRODUCTION ENVIRONMENT
+#
+# To create a pre-compiled cached version and make it available on production,
+# the script has to be ran from a builder (build-default or build-large) and
+# it's required to set the following environment variables for an IAM user with
+# permissions on ``build-tools`` S3's bucket:
+#
+#   AWS_ACCESS_KEY_ID
+#   AWS_SECRET_ACCESS_KEY
+#   AWS_ENDPOINT_URL
+#
+#
+# USAGE
+#
+#  ./scripts/compile_version_upload.sh python 3.9.6
+#
+#
+
+set -e
+
+# Define variables
+SLEEP=350
+OS="ubuntu-20.04"
+TOOL=$1
+VERSION=$2
+
+# https://stackoverflow.com/questions/59895/how-can-i-get-the-source-directory-of-a-bash-script-from-within-the-script-itsel
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+# Spin up a container with the Ubuntu 20.04 LTS image
+CONTAINER_ID=$(docker run --user docs --rm --detach --volume ${SCRIPT_DIR}/python-build.diff:/tmp/python-build.diff readthedocs/build:$OS sleep $SLEEP)
+echo "Running all the commands in Docker container: $CONTAINER_ID"
+
+# Install the tool version requested
+if [[ $TOOL == "python" ]]
+then
+    # Download list for all Python to force pyenv to be installed
```

Comment:
Why can't we just do `asdf install python`? Might be useful for a comment here explaining it.

Reply:
This chunk of code was removed. It was a small hack to download the file required to patch.